modules/storages/app/common/storages/peripherals/oauth_configurations/configuration_interface.rb

```diff
@@ -41,6 +43,21 @@ def to_httpx_oauth_config = raise ::Storages::Errors::SubclassResponsibility
         def authorization_uri(state: nil)
           basic_rack_oauth_client.authorization_uri(scope:, state:)
         end
+
+        def extract_origin_user_id(oauth_client_token)
+          auth_strategy = Registry
+                            .resolve("#{storage_name}.authentication.user_bound")
+                            .call(user: oauth_client_token.user)
+          Registry
+            .resolve("#{storage_name}.queries.user")
+            .call(auth_strategy:, storage: @storage)
+            .match(
+              on_success: ->(user) { user[:id] },
+              on_failure: ->(error) { raise "UserQuery responed with #{error}" }
+            )
+        end
+
+        def storage_name = raise ::Storages::Errors::SubclassResponsibility
```

Comment:
Defining this method might not be necessary. As far as I can tell, we always have a `@storage` when making these calls. At some point `Storage#to_s` was rewritten to return the provider name short form that can be used in registry resolve calls.

So what you should be able to do is `resolve("#{@storage}.your.stuff.here")`.

Reply:
Yes, true. That's absolutely unnecessary. Thanks!